### ISSUE TITLE
Docsy vs Asciidoctor writing standards for PlantUML blocks

### ISSUE THREAD

**Issue by adrianovieira** [google/docsy#761]
Hi folks! 

I'm using Docsy for a long time now, and I usually write the documents in *Asciidoc* format.

The issue is that *Asciidoc* standard to write PlantUML block is `[plantuml]`, e.g.:

```
.Sequence Diagram (ascidoc strandard)
[plantuml]
----
@startuml
Alice -> Bob: Authentication Request
Bob --> Alice: Authentication Response

Alice -> Bob: Another authentication Request
Alice <-- Bob: Another authentication Response
@enduml
----
```

But for Hugo+Docsy to work we need to write it in a slightly different standard `[source, plantuml]`, for example:

```
.Sequence Diagram (hugo+docsy strandard)
[source, plantuml]
----
@startuml
Alice -> Bob: Authentication Request
Bob --> Alice: Authentication Response

Alice -> Bob: Another authentication Request
Alice <-- Bob: Another authentication Response
@enduml
----
```

Is there a workaround that I can do here to get this working in the same standard as *Asciidoc* (`[plantuml]`)?

**Comment by LisaFC** [comment]
@akojimsg, any ideas (as you added the Plantuml support)?

**Comment by andreaslongo** [comment]
Asciidoctor has good support for diagrams with the `asciidoctor-diagram` plugin.
As a workaround, you could let Asciidoctor handle the diagrams instead of Docsy.

Have a look at my example repository to get an idea of how this works:

- https://github.com/andreaslongo/asciidoctor-diagram-hugo-docsy

TLDR: You need a working [Asciidoctor Diagram](https://docs.asciidoctor.org/diagram-extension/latest/) plugin and some settings in your [Hugo config](https://github.com/andreaslongo/asciidoctor-diagram-hugo-docsy/blob/400041ffeee7a007f99a59d2c5a326e7e762b21e/config.yaml#L5-L10). Then run `hugo --destination public` to generate your site.


I've included some sample HTML output from Asciidoctor. This may or may not be useful to streamline Docsy's diagram syntax with Asciidoctor:

- https://andreaslongo.github.io/asciidoctor-diagram-hugo-docsy/diagram/#_create_diagrams

**Comment by adrianovieira** [comment]
Hi, @andreaslongo ! 
thx

We are documenting systems using Asciidoc format for some time now, and publishing then on intranet sites using Hugo+Docsy.
I did it my way for a demo site as you all could notice the results in this link https://adrianovieira.gitlab.io/plantuml-demo

The trouble is that _Hugo+Docsy_ is working in a different standard (`[source, plantuml]`) when compared to _Asciidoctor_ standard (`[plantuml]`).

I'm not really sure, but in my opinion _Docsy_ could adopt the same _Asciidoctor_ standard.
What do you all think about this?

**Comment by chalin** [comment]
- Pre-0.17.0 triage campaign: #2717
- Out of scope: the Asciidoctor-side solution demonstrated in-thread is the right home.
